### PR TITLE
chore: (multi-domain) updating experimental flag to experimentalLoginFlows

### LINKED ISSUE #20823: [Multi-domain]: Create experimentalLoginFlows flag
### What would you like?

* Rename the `experimentalMultiDomain` to `experimentalLoginFlows`
* Update where the `experimentalSessionSupport` flag is being used and instead use `experimentalLoginFlows`
* Remove the `experimentalSessionSupport` flag.

### Why is this needed?

Session support is required to enable multi-domain.

### PR DESCRIPTION
**Pull request by mschile**
- Closes #20823

### User facing changelog

n/a

### Additional details

* Combined `experimentalSessionSupport` and `experimentalMultiDomain` to `experimentalLoginFlows`
  * Displays console warning if `experimentalSessionSupport` is enabled in the global config
  * Displays error if user tries to use `cy.session` with `experimentalSessionSupport` enabled
* Updated all tests to support sessions being turned on
* Updated `session_spec` system-test to use `127.0.0.1` instead of `127.0.0.2` since mac os doesn't loopback on `127.0.0.2`

### How has the user experience changed?

Console log warning for global config:

Screen Shot 2022-04-06 at 10 44 49 AM

Test failure if `experimentalSessionSupport` enabled and `cy.session` is used:
Screen Shot 2022-04-06 at 10 48 04 AM

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [x] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by chrisbreiding — APPROVED** [review]

File: packages/driver/src/cy/commands/sessions.ts [hunk 1]
```diff
@@ -278,8 +278,12 @@ export default function (Commands, Cypress, cy) {
   }
 
   function throwIfNoSessionSupport () {
-    if (!Cypress.config('experimentalSessionSupport')) {
-      $errUtils.throwErrByPath('sessions.experimentNotEnabled')
+    if (!Cypress.config('experimentalLoginFlows')) {
```
Comment (mschile) [inline packages/driver/src/cy/commands/sessions.ts:281]: Should I fail the test if `experimentalSessionSupport` is enabled regardless of `experimentalLoginFlows`?

File: packages/driver/cypress/integration/e2e/multi-domain/basic_login_spec.ts [hunk 2]
```diff
@@ -94,8 +93,10 @@ describe('basic login', { experimentalSessionSupport: true }, () => {
         cy.visit('http://www.foobar.com:3500/fixtures/auth/index.html')
       })
 
+      // @ts-ignore
       // TODO: Add createDomain primary domain config does not exist yet
       it.skip('D logs in with primary set via switch to domain config', { baseUrl: undefined, primaryDomain: 'localhost' }, () => {
+        // @ts-ignore
         cy.createDomain('http://idp.com:3500', () => { // PrimaryDomain set to localhost
```
Comment (mjhenkes) [inline packages/driver/cypress/integration/e2e/multi-domain/basic_login_spec.ts:100]: I should clean up this test file in another PR. I could probably either delete this test or switch it over to cy.origin

File: packages/driver/src/cy/commands/sessions.ts (hunk 1 above)
Comment (mjhenkes) [reply packages/driver/src/cy/commands/sessions.ts:281]: No, it doesn't hurt anything if they're both enabled, besides extraneous config, but that doesn't' seem like a fail-able offense.

File: packages/driver/src/cypress.ts [hunk 3]
```diff
@@ -281,7 +258,7 @@ class $Cypress {
       return null
     }
 
-    this.Cookies = $Cookies.create(config.namespace, d)
+    this.Cookies = $Cookies.create(config.namespace, domainName)
```
Comment (mjhenkes) [inline packages/driver/src/cypress.ts:261]: 👍

File: packages/driver/src/cypress/error_messages.ts [hunk 4]
```diff
@@ -1531,9 +1531,19 @@ export default {
...
     },
-    experimentNotEnabled: {
-      message: 'experimentalSessionSupport is not enabled. You must enable the experimentalSessionSupport flag in order to use Cypress session commands',
-      docsUrl: 'https://on.cypress.io/session',
+    experimentNotEnabled (experimentalSessionSupport) {
+      if (experimentalSessionSupport) {
+        return {
+          message: stripIndent`
+          ${cmd('session')} requires enabling the \`experimentalLoginFlows\` flag. The \`experimentalSessionSupport\` flag was enabled but was removed in Cypress version 9.6.0. Please see the migration guide for updating.`,
+          docsUrl: 'https://on.cypress.io/migration-guide',
```
Comment (mjhenkes) [inline packages/driver/src/cypress/error_messages.ts:1539]: We will need to remember to add this to the migration guide.

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**